scheduler/test/cook/test/kubernetes/controller.clj

```diff
@@ -24,6 +26,57 @@
       (is (= :reason-invalid-offers
              (controller/container-status->failure-reason {:name "test-cluster"} "12345"
                                                           pod-status container-status))))))
+(deftest test-process
+  (let [do-process (fn [cook-expected-state existing]
+                     (let [name "name"
+                           result
+                           (controller/process
+                             {:api-client nil
+                              :k8s-actual-state-map (atom {name {:cook-expected-state cook-expected-state}})
+                              :cook-expected-state-map (atom {name {:synthesized-state existing :pod nil}})}
+                             name)]
+                       (get result name)))]
+    (with-redefs [controller/delete-pod  (fn [_ _] nil)
+                  controller/kill-pod  (fn [_ cook-expected-state-dict _] cook-expected-state-dict)
+                  controller/launch-pod (fn [_ cook-expected-state-dict] cook-expected-state-dict)
+                  controller/log-weird-state (fn [_ _] :illegal)
+                  controller/pod-has-just-completed (fn [_] {:cook-expected-state :cook-expected-state/completed})
+                  controller/write-status-to-datomic (fn [_] :illegal)]
+
+      (is (= nil) (do-process :cook-expected-state/completed :missing))
+      (is (= nil) (do-process :cook-expected-state/completed :pod/succeeded))
```

Comment:
These are also outside the `nil`. We should use `nil?`.

Reply:
Ok.